src/core/border_router/routing_manager.cpp

```diff
@@ -632,6 +658,18 @@ void RoutingManager::EvaluateRoutingPolicy(void)
 
     SendRouterAdvertisement(kAdvPrefixesFromNetData);
 
+#if OPENTHREAD_CONFIG_SRP_SERVER_ENABLE
+    if (Get<Srp::Server>().IsAutoEnableMode())
+    {
+        // If SRP server uses the auto-enable mode, we enable the SRP
+        // server on the first RA transmission when we have decided
+        // on OMR and on-link prefixes. Note that if SRP server is
+        // already enabled, calling `Enable()` again does nothing.
+
+        Get<Srp::Server>().Enable();
```

Comment:
```suggestion
    if (Get<Srp::Server>().IsAutoEnableMode() && !mRsSender.IsInProgress())
    {
        // If SRP server uses the auto-enable mode, we enable the SRP
        // server on the first RA transmission when we have decided
        // on OMR and on-link prefixes. Note that if SRP server is
        // already enabled, calling `Enable()` again does nothing.

        Get<Srp::Server>().Enable();
```

`RoutingManager::EvaluateRoutingPolicy` can be called when we are sending RS messages and the on-link prefix may not be determined yet, so we need to ensure that we are not sending RS. It's better if we have a method `hasFavoredOnLinkPrefix()` to tell either there is a discovered on-link prefix or we are advertising our own (so that the connectivity from End Device to Infra link is established).

Reply:
Good point!  Thanks. 
- I remembered that we `VerifyOrExit(!mRsSender.IsInProgress());` check and I thought it was at `EvaluateRoutingPolicy()`level  but it is only in `EvaluateOnLinkPrefix()` (only skipping on-link).
- I was actually initially thinking of adding something similar (check if we have decided on on-link prefix or not).

Reply:
Updated this in new push. Added `IsInitalPolicyEvaluationDone()` method to do the check and use this to decide when/if to enable/disable SRP server.

Reply:
Thanks! `IsInitalPolicyEvaluationDone()` looks good.